sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeFormatterHelper.scala

```diff
@@ -87,7 +92,9 @@ trait DateTimeFormatterHelper {
 private object DateTimeFormatterHelper {
   val cache = CacheBuilder.newBuilder()
     .maximumSize(128)
-    .build[(String, Locale), DateTimeFormatter]()
+    .build[(String, Locale, Boolean), DateTimeFormatter]()
+
+  final val extractor = "^([^S]*)(S*)(.*)$".r
```

Comment:
One thing I'm worried about is ambiguity. e.g. `ss.SSSyyyy`, how are we going to parse `11.12020`? I think we can only support var length if the parser after `SSS` is not number type.

Reply:
A simplification is to only support var-length if the part after `SSS` is timezone (a few chars like xXzZ etc.)